### ISSUE TITLE
Misleading HINT when you hit `neon.max_cluster_size` limit

### ISSUE THREAD

**Issue by hlinnaka** [neondatabase/neon#3910]
```
ERROR:  could not extend file because cluster size limit (10240 MB) has been exceeded
HINT:  This limit is defined by neon.max_cluster_size GUC
```
That's technically accurate, but very misleading to a user because they cannot change the `neon.max_cluster_size` setting.

**Comment by problame** [comment]
Related to this, when changing the limit in the Admin console, I found that in order for the setting to take effect, it is necessary to restart the compute.
I did that using the management API's maintenance mode.

**Comment by vadim2404** [comment]
@sharnoff , if I'm not mistaken, you did something related. Am I right?

**Comment by sharnoff** [comment]
Ah, yup - this was resolved in #4173, should have been closed then.
